packages/flame/lib/src/collision_detection/broadphase.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:meta/meta.dart';
+
+import '../../collision_detection.dart';
+
+/// The [Broadphase] class is used to make collision detection more efficient
+/// by doing a rough estimation of which hitboxes that can collide before their
+/// actual intersections are calculated.
+///
+/// Currently there is only one implementation of [Broadphase] and that is
+/// [Sweep].
+abstract class Broadphase<T extends Hitbox<T>> {
+  final List<T> items;
+
+  Broadphase({List<T>? items}) : items = items ?? [];
+
+  Set<Potential<T>> query();
+}
+
+/// A [Potential] is a tuple that is used to contain two potentially colliding
+/// hitboxes.
+@immutable
+class Potential<T> {
```

Comment:
`Pair` is more vague though, it should be a word that describes a pair in a potential collision.t
Maybe `CollisionProspect` or something like that?

Reply:
The way the class is written right now, it could be a completely generic utility class that lives somewhere in `utils/`. Or do you envision that we'll need some collision-specific features in here?

Reply:
I don't think we should create a general tuple class, there are reasons that the Dart team haven't added it, I think mainly because it easily leads to anti-patterns among folks who don't know what they are doing.